extensions/amp-bind/0.1/amp-state.js

```diff
@@ -154,10 +146,11 @@ export class AmpState extends AMP.BaseElement {
    */
   fetchAndUpdate_(isInit, opt_refresh) {
     const ampdoc = this.getAmpDoc();
-    return this.fetch_(ampdoc, this.element, isInit, opt_refresh)
-        .then(json => {
-          this.updateState_(json, isInit);
-        });
+    // Don't fetch in prerender mode.
+    const viewer = Services.viewerForDoc(this.getAmpDoc());
```

Comment:
fetchAndUpdate_ is also called in the registered refresh action method call. We should set the viewer service as a private property on the buildCallback so that we don't need to get the service everything time this method is called.

Reply:
I'd prefer not to do this optimization due to the readability and testing cost. I don't think it makes a perf difference in practice.